### PR TITLE
[4348] Sidebar event propagation between widgets seems not to work

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4348

### CODE REVIEW

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 1]
```diff
@@ -977,7 +977,7 @@
           markup = markup.replace(/\&gt;/g, '>');
           markup = markup.replace(/\&lt;/g, '<');
           markup = markup.replace(/\&amp;/g, '&');
-          markup = markup.replace(/\&#x27;/g, "'");
+          markup = markup.replace(/\&#x27;/g, '\'');
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: The `&#x27;` is the `'` character. Why this change?

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js [hunk 2]
```diff
@@ -1512,7 +1512,8 @@
                       treePrint(servPath);
                     }
                   },
-                  failure: function() {}
+                  failure: function() {
+                  }
```
Comment (rart) [inline static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: All these changes look like IntelliJ formatter

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 3]
```diff
@@ -329,7 +331,13 @@ export function ItemMenu(props: ItemMenuProps) {
       case 'paste': {
         paste(site, item.path).subscribe(
           () => {
-            dispatch(unSetClipBoard());
+            dispatch(batchActions([unSetClipBoard(), showPasteItemSuccessNotification()]));
+            hostToHost$.next({
+              type: 'ITEM_PASTED',
+              payload: {
+                path: item.path
+              }
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: let's go ahead and send the whole item

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx [hunk 4]
```diff
@@ -209,7 +209,7 @@ function NewContentDialogBody(props: NewContentDialogProps) {
     selectedContentType
   } = props;
   const [openSelector, setOpenSelector] = useState(false);
-  const defaultFilterType = 'all';
+  const defaultFilterType = (type === 'new') ? 'all' : item.systemType;
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/NewContentDialog.tsx]: I think we **may be** masking a backend bug. Perhaps we should roll this back and display strictly as the backend replies.

File: ui/app/src/modules/Content/Delete/DeleteDialog.tsx [hunk 5]
```diff
@@ -309,6 +311,14 @@ function DeleteDialogWrapper(props: DeleteDialogProps) {
     deleteItems(siteId, submissionComment, data).subscribe(
       (response) => {
         setApiState({ submitting: false });
+
+        hostToHost$.next({
+          type: 'ITEM_DELETED',
+          payload: {
+            paths: selectedItems
```
Comment (rart) [inline ui/app/src/modules/Content/Delete/DeleteDialog.tsx]: Can we send also like the main item?

File: ui/app/src/services/content.ts [hunk 6]
```diff
@@ -951,7 +951,7 @@ export function getChildrenByPath(
     map(({ item }) =>
       Object.assign(parseLegacyItemToSandBoxItem(item.children), {
         parent: parseLegacyItemToSandBoxItem(item),
-        ...(item.children[0].contentType === '/component/level-descriptor' && {
+        ...(item.children[0]?.contentType === '/component/level-descriptor' && {
           levelDescriptor: {
```
Comment (rart) [inline ui/app/src/services/content.ts]: - This prop should always be there. If the level descriptor isn't there, leave it in as null.
- Perhaps we should be finding the item instead of assuming is the first.

File: ui/app/src/utils/i18n-legacy.ts [hunk 7]
```diff
@@ -1396,9 +1396,17 @@ export const itemSuccessMessages = defineMessages({
...
+    id: 'item.pasted',
+    defaultMessage: 'Item pasted successfully'
+  },
   itemReverted: {
     id: 'item.edited',
     defaultMessage: 'Item reverted successfully'
+  },
+  itemContentTypeChanged: {
+    id: 'item.contentTypeChanged',
+    defaultMessage: 'Item content type changed successfully'
```
Comment (rart) [inline ui/app/src/utils/i18n-legacy.ts]: item ~content~ type changed successfully

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js (hunk 1 above)
Comment (aachavarria) [reply static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: Prettier

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx (hunk 4 above)
Comment (aachavarria) [reply ui/app/src/modules/Content/Authoring/NewContentDialog.tsx]: I don't think so, because the BE service is for creating content in the selected path, not for changing the selected path contentType

File: static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js (hunk 2 above)
Comment (aachavarria) [reply static-assets/components/cstudio-contextual-nav/wcm-site-dropdown-mods/wcm-root-folder.js]: looks like there was a issue with prettier, will run it again

File: ui/app/src/components/Dialogs/BulkUploadDialog.tsx [hunk 8]
```diff
@@ -641,6 +651,10 @@ export default function BulkUploadDialog(props: BulkUploadProps) {
     e.preventDefault();
   };
 
+  const onFileUploaded = (path: string) => {
+    dispatch(systemEvent(fileUploaded({ target: path })));
+  };
```
Comment (rart) [inline ui/app/src/components/Dialogs/BulkUploadDialog.tsx]: Will trigger effect on every render

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 9]
```diff
@@ -240,15 +236,13 @@ export function ItemMenu(props: ItemMenuProps) {
...
+              showChangeContentTypeDialog({
                 item,
-                type: 'change',
+                rootPath: getRootPath(item.path),
                 selectedContentType: item.contentTypeId,
                 onContentTypeSelected: batchActions([
-                  closeNewContentDialog(),
-                  changeContentType({ contentTypeId: item.contentTypeId, path })
+                  closeChangeContentTypeDialog(),
+                  changeContentType({ contentTypeId: item.contentTypeId, path: item.path })
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: Can we have `newContentTypeId` and `originalContentTypeId`?

File: ui/app/src/services/content.ts [hunk 10]
```diff
@@ -1002,7 +1002,7 @@ export function duplicate(site: string, path: string): Observable<string> {
   return copy(site, path).pipe(
     switchMap(() => paste(site, getParentPath(parentPath))),
     // Duplicate only does shallow copy, so return the copied path
-    pluck('status', '0')
+    map((paths) => paths[0])
```
Comment (rart) [inline ui/app/src/services/content.ts]: `pluck('0')`

File: ui/app/src/state/actions/dialogs.ts [hunk 11]
```diff
@@ -200,10 +210,23 @@ export const pathSelectionDialogClosed = createAction('PATH_SELECTION_CLOSED');
...
+export const showCutItemSuccessNotification = createAction('SHOW_CUT_ITEM_SUCCESS_NOTIFICATION');
+export const showPasteItemSuccessNotification = createAction(
+  'SHOW_PASTE_ITEM_SUCCESS_NOTIFICATION'
+);
+export const showDuplicatedItemSuccessNotification = createAction(
+  'SHOW_DUPLICATED_ITEM_SUCCESS_NOTIFICATION'
+);
+export const showRevertItemSuccessNotification = createAction(
+  'SHOW_REVERT_ITEM_SUCCESS_NOTIFICATION'
+);
```
Comment (rart) [inline ui/app/src/state/actions/dialogs.ts]: Let's group all these system stuff under the new systemEvents actions file, only that, rename actions/systemEvents.ts to actions/system.ts

File: ui/app/src/state/actions/systemEvents.ts [hunk 12]
```diff
@@ -0,0 +1,40 @@
+/*
```
Comment (rart) [inline ui/app/src/state/actions/systemEvents.ts]: Rename to system.ts

File: ui/app/src/state/actions/systemEvents.ts [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+
+export const itemDeleted = createAction<{ targets: string[] }>('ITEM_DELETED');
+
+export const fileUploaded = createAction<{ target: string }>('FILE_UPLOADED');
+
+export const itemDuplicated = createAction<{ target: string; resultPath: string }>(
+  'ITEM_DUPLICATED'
+);
+
+export const systemEvent = createAction<StandardAction>('SYSTEM_EVENT');
```
Comment (rart) [inline ui/app/src/state/actions/systemEvents.ts]: let's call this one emitSystemEvent

File: ui/app/src/components/ItemMenu/ItemMenu.tsx (hunk 9 above)
Comment (aachavarria) [reply ui/app/src/components/ItemMenu/ItemMenu.tsx]: yes

File: ui/app/src/state/epics/dialogs.ts [hunk 14]
```diff
@@ -117,76 +120,139 @@ export default [
...
+              count: payload?.children.length ?? 1
+            })
+          })
+        );
+      }),
+      ignoreElements()
+    ),
+  (action$, state$, { intlRef: { current: intl } }: { intlRef: { current: IntlShape } }) =>
+    action$.pipe(
+      ofType(showCutItemSuccessNotification.type),
```
Comment (rart) [inline ui/app/src/state/epics/dialogs.ts]: Let's add all these to epics/system.ts

File: ui/app/src/state/epics/misc.ts [hunk 15]
```diff
@@ -15,65 +15,77 @@
...
-            showEditItemSuccessNotification,
-            pathNavigatorItemActionSuccess({ id: payload.id, option: 'refresh' })
-          ])
-        }))
-      );
-    }
-    return NEVER;
-  })
-);
+const systemEventPropagate: Epic = (action$) =>
```
Comment (rart) [inline ui/app/src/state/epics/misc.ts]: ==> epics/system.ts

**Review by rart — CHANGES_REQUESTED** [review]
Please address comments and fix conflicts

**Review by rart — APPROVED** [review]

File: static-assets/components/cstudio-forms/template-editor.js [hunk 16]
```diff
@@ -65,7 +65,7 @@ CStudioAuthoring.Module.requireModule(
                 },
                 'transform-path-to-url': {
                   label: 'Transform PATH to URL',
-                  value: '${urlTransformationService.transform(\'storeUrlToRenderUrl\', STOREURL)}'
+                  value: "${urlTransformationService.transform('storeUrlToRenderUrl', STOREURL)}"
```
Comment (rart) [inline static-assets/components/cstudio-forms/template-editor.js]: revert.
something seemed to have happened with this file. seeing to many quote changes that go against the format setting. any idea what’s up?

File: ui/app/src/components/Dialogs/BulkUploadDialog.tsx [hunk 17]
```diff
@@ -641,6 +649,13 @@ export default function BulkUploadDialog(props: BulkUploadProps) {
     e.preventDefault();
   };
 
+  const onFileUploaded = useCallback(
+    (path: string) => {
+      dispatch(emitSystemEvent(fileUploaded({ target: path })));
```
Comment (rart) [inline ui/app/src/components/Dialogs/BulkUploadDialog.tsx]: Do we also have a fileCreated? Is there a difference between upload and creation? Perhaps we should only leave the fileCreated

File: ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx [hunk 18]
```diff
@@ -31,35 +31,51 @@ import {
...
 import Suspencified, { SuspenseWithEmptyState } from '../../SystemStatus/Suspencified';
-import { withIndex, withoutIndex } from '../../../utils/path';
+import { getParentPath, withIndex, withoutIndex } from '../../../utils/path';
 import { useStyles } from './styles';
 import { translations } from './translations';
 import Header from './PathNavigatorHeader';
 import Breadcrumbs from './PathNavigatorBreadcrumbs';
 import NavItem from './PathNavigatorItem';
 import Nav from './PathNavigatorList';
-import ContentLoader from 'react-content-loader';
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx:41]: yarn remove it too

File: ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx [hunk 19]
```diff
@@ -31,35 +31,51 @@ import {
...
+  folderCreated,
+  folderRenamed,
+  itemCreated,
+  itemDuplicated,
+  itemsDeleted,
+  itemsPasted,
+  itemUpdated
+} from '../../../state/actions/system';
+import List from '@material-ui/core/List';
+import { PathNavigatorSkeletonItem } from './PathNavigatorSkeletonItem';
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx]: Should be default export

File: ui/app/src/models/GlobalState.ts [hunk 20]
```diff
@@ -98,7 +99,11 @@ export interface GlobalState {
       byPath: LookupTable<DetailedItem>;
       permissionsByPath: LookupTable<LookupTable<boolean>>;
     };
-    clipboard: string;
+    clipboard: {
+      type: 'cut | copy';
+      paths: string[];
+      sourceRootPath: string;
```
Comment (rart) [inline ui/app/src/models/GlobalState.ts]: This is the path where things were copied/cutter from, right? If so, let’s call it `sourcePath`.

Also, please move the workflow check from cut to paste – when type is cut.

File: ui/app/src/modules/Content/History/HistoryDialog.tsx [hunk 21]
```diff
@@ -53,6 +52,7 @@ import Dialog from '@material-ui/core/Dialog';
 import { batchActions } from '../../../state/actions/misc';
 import { fetchUserPermissions } from '../../../state/actions/content';
 import { asDayMonthDateTime } from '../../../utils/datetime';
+import { makeStyles } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/History/HistoryDialog.tsx]: fix core import

File: ui/app/src/modules/Content/History/VersionList.tsx [hunk 22]
```diff
@@ -30,11 +29,12 @@ import { LegacyVersion } from '../../../models/Version';
 import clsx from 'clsx';
 import palette from '../../../styles/palette';
 import LookupTable from '../../../models/LookupTable';
+import { makeStyles } from '@material-ui/core';
```
Comment (rart) [inline ui/app/src/modules/Content/History/VersionList.tsx]: core import

File: ui/app/src/state/actions/dialogs.ts [hunk 23]
```diff
@@ -101,6 +103,16 @@ export const closeNewContentDialog = createAction<StandardAction>('CLOSE_NEW_CON
...
 // endregion
 
+// region Change ContentType
+export const showChangeContentTypeDialog = createAction<Partial<ChangeContentTypeDialogStateProps>>(
+  'SHOW_CHANGE_CONTENT_TYPE_DIALOG'
+);
+export const closeChangeContentTypeDialog = createAction<StandardAction>(
+  'CLOSE_CHANGE_CONTENT_TYPE_DIALOG'
+);
+export const ChangeContentTypeDialogClosed = createAction('CHANGE_CONTENT_TYPE_DIALOG_CLOSED');
```
Comment (rart) [inline ui/app/src/state/actions/dialogs.ts:99]: name out of pattern/convention

File: ui/app/src/state/reducers/content.ts [hunk 24]
```diff
@@ -83,7 +101,9 @@ const reducer = createReducer<ContentState>(initialState, {
       ...state.items,
       permissionsByPath: {
         ...state.items.permissionsByPath,
-        [payload.path]: createPresenceTable(payload.permissions.map(value => value.replaceAll(' ', '_').replace(/-/g, '_')))
+        [payload.path]: createPresenceTable(
+          payload.permissions.map((value) => value.replaceAll(' ', '_').replace(/-/g, '_'))
```
Comment (rart) [inline ui/app/src/state/reducers/content.ts]: Consolidate into a single replace with regex

**Review by rart — CHANGES_REQUESTED** [review]

File: static-assets/components/cstudio-forms/template-editor.js (hunk 16 above)
Comment (rart) [reply static-assets/components/cstudio-forms/template-editor.js]: ha! Found it: https://prettier.io/docs/en/next/rationale.html#strings
Hadn't seen this until now, perhaps is recent.

File: ui/app/src/components/Dialogs/BulkUploadDialog.tsx (hunk 17 above)
Comment (aachavarria) [reply ui/app/src/components/Dialogs/BulkUploadDialog.tsx]: will change to use fileCreated

File: ui/app/src/models/GlobalState.ts (hunk 20 above)
Comment (aachavarria) [reply ui/app/src/models/GlobalState.ts]: it is the rootPath it is used to check if the itemMenu should show the paste option;

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 25]
```diff
@@ -243,20 +237,7 @@ export function ItemMenu(props: ItemMenuProps) {
         break;
       }
       case 'cut': {
-        cut(site, item).subscribe(
-          ({ success }) => {
-            if (success) {
-              dispatch(batchActions([setClipBoard({ path: item.path }), showCopyItemSuccessNotification()]));
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx:249]: Showing copy message instead of cut

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 26]
```diff
@@ -243,20 +237,7 @@ export function ItemMenu(props: ItemMenuProps) {
...
-          },
-          (response) => {
-            dispatch(
-              showErrorDialog({
-                error: response
-              })
-            );
-          }
-        );
+        dispatch(itemCut({ path }));
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: Should be dispatched on success

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 27]
```diff
@@ -303,18 +296,22 @@ export function ItemMenu(props: ItemMenuProps) {
         break;
       }
       case 'paste': {
-        paste(site, item.path).subscribe(
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx:306]: Don't see the paste operation getting executed at all if this is removed. Looks like you're dispatching the paste system event before the paste is actually performed

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 28]
```diff
@@ -425,10 +422,7 @@ export function ItemMenu(props: ItemMenuProps) {
           showCreateFileDialog({
             path: withoutIndex(item.path),
             type: 'template',
-            onCreated: batchActions([
-              closeCreateFileDialog(),
-              onItemMenuActionSuccessCreator?.({ item, option: 'refresh' })
-            ])
+            onCreated: batchActions([closeCreateFileDialog()])
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: No longer batch

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 29]
```diff
@@ -438,10 +432,7 @@ export function ItemMenu(props: ItemMenuProps) {
           showCreateFileDialog({
             path: withoutIndex(item.path),
             type: 'controller',
-            onCreated: batchActions([
-              closeCreateFileDialog(),
-              onItemMenuActionSuccessCreator?.({ item, option: 'refresh' })
-            ])
+            onCreated: batchActions([closeCreateFileDialog()])
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: No longer batch

File: ui/app/src/components/ItemMenu/ItemMenu.tsx [hunk 30]
```diff
@@ -471,7 +462,7 @@ export function ItemMenu(props: ItemMenuProps) {
           showUploadDialog({
             path: item.path,
             site,
-            onClose: batchActions([closeUploadDialog(), onItemMenuActionSuccessCreator?.({ item, option: 'upload' })])
+            onClose: batchActions([closeUploadDialog()])
```
Comment (rart) [inline ui/app/src/components/ItemMenu/ItemMenu.tsx]: No longer batch

There's a bunch of these. Please fix all.

File: ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx [hunk 31]
```diff
@@ -369,15 +427,13 @@ export default function PathNavigator(props: WidgetProps) {
...
         onSimpleMenuClick={onSimpleMenuClick}
-        onItemMenuActionSuccessCreator={onItemMenuActionSuccessCreator}
       />
     </Suspencified>
   );
 }
 
 export function PathNavigatorUI(props: WidgetUIProps) {
   const classes = useStyles();
-  // region consts {...} = props
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx]: Leave region unless it ends up being a single line. If the region is removed, the endregion should be too.

File: ui/app/src/components/Navigation/PathNavigator/PathNavigatorSkeletonItem.tsx [hunk 32]
```diff
@@ -0,0 +1,34 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useStyles } from './styles';
+import ListItem from '@material-ui/core/ListItem';
+import { Skeleton } from '@material-ui/lab';
```
Comment (rart) [inline ui/app/src/components/Navigation/PathNavigator/PathNavigatorSkeletonItem.tsx]: Check all lab imports to be loaded from its individual file (e.g. material/lab/skeleton)

File: ui/app/src/modules/Content/Authoring/NewContentDialog.tsx [hunk 33]
```diff
@@ -158,58 +121,32 @@ export interface NewContentDialogStateProps extends NewContentDialogBaseProps {
...
 export default function NewContentDialog(props: NewContentDialogProps) {
   return (
     <Dialog open={props.open} onClose={props.onClose} fullWidth maxWidth="md">
-      <NewContentDialogBody {...props} />
+      <NewContentDialogWrapper {...props} />
     </Dialog>
   );
 }
 
-function NewContentDialogBody(props: NewContentDialogProps) {
```
Comment (rart) [inline ui/app/src/modules/Content/Authoring/NewContentDialog.tsx:191]: Looks like the merge resulted in some changes getting lost. These renames should persist. ...DialogBody is the correct name.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/ItemMenu/ItemMenu.tsx (hunk 27 above)
Comment (aachavarria) [reply ui/app/src/components/ItemMenu/ItemMenu.tsx:306]: itemCut and itemPasted on this case arent actions from systemEvents;

File: ui/app/src/components/ToolsPanelPage/ToolsPanelPage.tsx [hunk 34]
```diff
@@ -18,6 +18,8 @@ import React, { useContext } from 'react';
 import ToolPanel from '../../modules/Preview/Tools/ToolPanel';
 import { renderWidgets, TempTestContext } from '../Widget/Widget';
 
+//console.log(formatMessage({ id: 'words.cancel', defaultMessage: 'words.cancel' }));
```
Comment (rart) [inline ui/app/src/components/ToolsPanelPage/ToolsPanelPage.tsx]: Should not commit

File: ui/app/src/models/GlobalState.ts [hunk 35]
```diff
@@ -115,6 +122,7 @@ export interface GlobalState {
     currentUrl: string;
     computedUrl: string;
     showToolsPanel: boolean;
+    //toolsPanelPageStack>>pages[]
```
Comment (rart) [inline ui/app/src/models/GlobalState.ts]: should not commit

File: ui/app/src/models/GlobalState.ts [hunk 36]
```diff
@@ -178,6 +187,7 @@ export interface GlobalState {
     error: ApiResponse;
     isFetching: boolean;
     currentSite: string;
+    //preview>toolsPanel>widgets
```
Comment (rart) [inline ui/app/src/models/GlobalState.ts]: should not commit

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/CrafterCMSNextBridge.tsx [hunk 37]
```diff
@@ -104,3 +107,13 @@ function setUpLocaleChangeListener(update: Function, currentIntl: IntlShape) {
   document.addEventListener('setlocale', handler, false);
   return () => document.removeEventListener('setlocale', handler, false);
 }
+
+function SnackbarCloseButton({ id }) {
```
Comment (rart) [inline ui/app/src/components/CrafterCMSNextBridge.tsx]: Move to it's own folder/file

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/SnackbarCloseButton/SnackbarCloseButton.tsx [hunk 38]
```diff
@@ -0,0 +1,30 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useSnackbar } from 'notistack';
+import { IconButton } from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import React from 'react';
+
+export default function SnackbarCloseButton({ id }) {
```
Comment (rart) [inline ui/app/src/components/SnackbarCloseButton/SnackbarCloseButton.tsx]: Props type please

**Review by rart — APPROVED** [review]

File: ui/app/src/components/SnackbarCloseButton/SnackbarCloseButton.tsx (hunk 38 above)
Comment (rart) [reply ui/app/src/components/SnackbarCloseButton/SnackbarCloseButton.tsx]: Done.

**Review by rart — APPROVED** [review]
Conflicts resolved.
